### PR TITLE
Fix BM side comment permission 

### PR DESCRIPTION
**Pull request by cedrtang**
To test, need to add the correct permission to participants

### CODE REVIEW

File: assembl/graphql/post.py [hunk 1]
```diff
@@ -764,11 +764,13 @@ class Input:
...
     @abort_transaction_on_exception
     def mutate(root, args, context, info):
-        require_cls_permission(CrudPermissions.CREATE, models.Extract, context)
         discussion_id = context.matchdict['discussion_id']
-
         user_id = context.authenticated_userid or Everyone
 
+        permissions = get_permissions(user_id, discussion_id)
+        if P_ADD_SIDE_COMMENT not in permissions and require_cls_permission(CrudPermissions.CREATE, models.Extract, context):
+            raise HTTPUnauthorized()
```
Comment (ayyazdaniaryan) [inline assembl/graphql/post.py]: We try to use the graphql `permission_helpers` as much as possible. For a start, there is a more descriptive error that is translated across multiple languages that is passed through. Also, you can use the `require_instance_permission` I suspect too (double check that).

Comment (cedrtang) [reply assembl/graphql/post.py]: @ayyazdaniaryan `require_instance_permission` is used for existing instances obviously so can only be used for updating/deleting, not creating. `permissions_helpers` doesn't handle single permissions, only class and instances. I could add a new function for this.

Comment (ayyazdaniaryan) [reply assembl/graphql/post.py]: You're right. Nah, this is simple enough to check. If the use-case grows over the APIs, we'll make it a function.

File: assembl/models/preferences.py [hunk 2]
```diff
@@ -807,12 +809,14 @@ def get_preference_data_list(self):
...
                     P_SEND_SYNTHESIS,
                     P_VOTE,
+                    P_ADD_SIDE_COMMENT,
                 ],
                 R_PARTICIPANT: [
                     P_ADD_POST,
                     P_DELETE_MY_POST,
                     P_EDIT_MY_POST,
                     P_VOTE,
+                    P_ADD_SIDE_COMMENT,
```
Comment (ayyazdaniaryan) [inline assembl/models/preferences.py]: Cool, since the base Participant will have this perm, I suggest the the `R_CATCHER` to also have this perm as well.

**Pull request merged**